controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -335,9 +336,20 @@ func recordActionFailure(info *reconcileInfo, errorType metal3v1alpha1.ErrorType
 	return actionFailed{dirty: true, ErrorType: errorType, errorCount: info.host.Status.ErrorCount}
 }
 
-func recordActionDelayed(info *reconcileInfo) actionResult {
+func recordActionDelayed(info *reconcileInfo, nextState metal3v1alpha1.ProvisioningState) actionResult {
+	var counter prometheus.Counter
+
+	switch nextState {
+	case metal3v1alpha1.StateInspecting, metal3v1alpha1.StateProvisioning:
+		counter = delayedProvisioningHostCounters.With(hostMetricLabels(info.request))
+	case metal3v1alpha1.StateDeprovisioning, metal3v1alpha1.StateDeleting:
+		counter = delayedDeprovisioningHostCounters.With(hostMetricLabels(info.request))
+	case metal3v1alpha1.StateReady:
```

Comment:
Assuming that the state parameter could be also the current one, this can be removed

Reply:
Similar questing about default value of counter.
https://github.com/metal3-io/baremetal-operator/pull/807#discussion_r601311963